File: golang/palindrome_linked_list.go
package golang

import (
	"golang/model"
)

// Space : O(n)
// Time  : O(n)

func isLLPalindrome(head *model.ListNode) bool {
	mem := []int{}

	for head != nil {
		mem = append(mem, head.Val)
		head = head.Next
	}

	s, e := 0, len(mem)-1
	for s < e {
		if mem[s] != mem[e] {
			return false
		}
		s++
		e--
	}

	return true
}

File: python3/palindrome_linked_list.py
from .model import ListNode

# Space : O(n)
# Time  : O(n)


class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        if not head:
            return True

        temp = []
        p = head
        while p:
            temp.append(p.val)
            p = p.next

        return temp == temp[::-1]
